en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
 }
 ```
 
+## Code style
+
+- Prioritise the human readability of code. Avoid "clever" code that aims to reduce the number of lines at the expense of readbability. This is particularly important when writing Javascript.
+- Avoid writing any new "magic" code as it is very hard to follow.
+- Within reason, favour larger files that have everything contained in them rather than multiple small files that require developers to jump around multiple files to follow a single code path.
+
+## Inline comments
+- If there is a non-obvious reason for a code operation, include verbose inline comments that explain *why* we are doing this operation. Remember that while something may be obvious to you now, it won't be obvious to someone else in 6 months.
+- Inline comments that explain *what* the code is doing are generally fine as it can make code easier to follow and is particularly useful for larger methods, though it may be better to simply split larger methods in to several private methods.
+
+## Naming conventions
+
+### Casing
+
+- Use `camelCase` for variables, method names and parameters
+- Use `UpperCamelCase` for class names, interfaces, traits and enums
+- Use `snake_case` for private static configuration properties.
+- Use `SCREAMING_SNAKE_CASE` for constants including class contants, and environment variables
+- Note that there will be some inconsistencies with legacy code where static methods have been a combination of `MyClass::camelCase()` and `MyClass::snake_case()`
+
+### Accessors
+
+- For accessors omit the leading 'get' with getters and prefix with 'set' for setters. e.g. `function myValue():sting { return $this->value; }` and `function setMyValue(string $value): void { $this->value = $value; }`.
+- Note that there will be some inconsistencies with legacy code as get accessors have been a combination of "with get", "without get", and "without get with the first letter uppercased"
+
+### File suffixes and prefixes
+
+- This is for new classes and interfaces that are created. Legacy Silvesrtripe code is inconsistent whether or not it follows these conventions.
+- Suffix the classnames with the type of class that is subclassed. The suffixed is sometimes a shortened version of the parent class because it reads better while retaining easy comprehension. Here are some common examples with the parents classes in brackets
+  - `Exception` (Exception)
+  - `Controller` (Controller)
+  - `Extension` (Extension, DataExtension)
+  - `Admin` (ModelAdmin)
+  - `Handler` (RequestHandler)
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
```

Comment:
Creating an AdminController is on the cards, and presumably we'd keep the forced authentication and $url_segment etc for easy routing.

From CMS 6 we'd want to change a bunch of the admin controllers that currently extend LeftAndMain to now extend AdminController.  In that instance I guess maybe it would make sense to stuff those subclasses with with "AdminController" rather than "Controller" since it's more descriptive?

Reply:
Have updated to say that subclasses of LeftAndMain intended to be in the CMS Menu should be suffixed with `Admin`